agent/proto/resourcesmonitor/fetcher.go

```diff
@@ -0,0 +1,61 @@
+package resourcesmonitor
+
+import (
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/cli/clistat"
+)
+
+type ResourcesFetcher interface {
+	FetchResourceMonitoredMemory() (total int64, used int64, err error)
+	FetchResourceMonitoredVolume(volume string) (total int64, used int64, err error)
+}
+
+type resourcesFetcher struct {
+	fetcher *clistat.Statter
```

Comment:
```suggestion
	*clistat.Statter
```

Creating a field called "fetcher" in a struct called "fetcher" is confusing. You can just make the struct just contain the Statter and add additonal methods.